code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -138,19 +138,29 @@ RESTRICT_TYPE(/datum/antagonist/traitor)
  * Create and assign a single randomized human traitor objective.
  */
 /datum/antagonist/traitor/proc/forge_single_human_objective()
+	var/datum/objective/objective_to_add
+
 	if(prob(50))
 		if(length(active_ais()) && prob(100 / length(GLOB.player_list)))
-			add_antag_objective(/datum/objective/destroy)
+			objective_to_add = /datum/objective/destroy
+
 		else if(prob(5))
-			add_antag_objective(/datum/objective/debrain)
+			objective_to_add = /datum/objective/debrain
+
 		else if(prob(30))
-			add_antag_objective(/datum/objective/maroon)
+			objective_to_add = /datum/objective/maroon
+
 		else if(prob(30))
-			add_antag_objective(/datum/objective/assassinateonce)
+			objective_to_add = /datum/objective/assassinateonce
+
 		else
-			add_antag_objective(/datum/objective/assassinate)
+			objective_to_add = /datum/objective/assassinate
 	else
-		add_antag_objective(/datum/objective/steal)
+		objective_to_add = /datum/objective/steal
+
+	if(delayed_objectives)
```

Comment:
Quick question to validate. Will this break admins adding a random new objective later in the round for this antag? Or is that applied differently?

Reply:
Mistakenly removed the line that set `delayed_objectives` to `FALSE`, added it back so it should work fully after the timer has gone off!